Mathlib/Combinatorics/Additive/Energy.lean

```diff
@@ -14,62 +16,77 @@ import Mathlib.Data.Fintype.Prod
+* `E[s, t]` for `Finset.addEnergy s t`.
+* `Eₘ[s, t]` for `Finset.mulEnergy s t`.
+* `E[s]` for `E[s, s]`.
+* `Eₘ[s]` for `Eₘ[s, s]`.
+
 ## TODO
 
 It's possibly interesting to have
 `(s ×ˢ s) ×ˢ t ×ˢ t).filter (fun x : (α × α) × α × α ↦ x.1.1 * x.2.1 = x.1.2 * x.2.2)`
-(whose `card` is `multiplicativeEnergy s t`) as a standalone definition.
+(whose `card` is `mulEnergy s t`) as a standalone definition.
 -/
 
-
-section
-
-variable {α : Type*} [PartialOrder α] {x y : α}
-
-end
+open scoped BigOperators Pointwise
 
 variable {α : Type*} [DecidableEq α]
 
 namespace Finset
-
 section Mul
-
 variable [Mul α] {s s₁ s₂ t t₁ t₂ : Finset α}
+
 /-- The multiplicative energy of two finsets `s` and `t` in a group is the number of quadruples
 `(a₁, a₂, b₁, b₂) ∈ s × s × t × t` such that `a₁ * b₁ = a₂ * b₂`. -/
-@[to_additive additiveEnergy
+@[to_additive
       "The additive energy of two finsets `s` and `t` in a group is the
       number of quadruples `(a₁, a₂, b₁, b₂) ∈ s × s × t × t` such that `a₁ + b₁ = a₂ + b₂`."]
-def multiplicativeEnergy (s t : Finset α) : ℕ :=
+def mulEnergy (s t : Finset α) : ℕ :=
   (((s ×ˢ s) ×ˢ t ×ˢ t).filter fun x : (α × α) × α × α => x.1.1 * x.2.1 = x.1.2 * x.2.2).card
-#align finset.multiplicative_energy Finset.multiplicativeEnergy
-#align finset.additive_energy Finset.additiveEnergy
-
-@[to_additive additiveEnergy_mono]
-theorem multiplicativeEnergy_mono (hs : s₁ ⊆ s₂) (ht : t₁ ⊆ t₂) :
-    multiplicativeEnergy s₁ t₁ ≤ multiplicativeEnergy s₂ t₂ :=
-  card_le_card <|
-    filter_subset_filter _ <|
-      product_subset_product (product_subset_product hs hs) <| product_subset_product ht ht
-#align finset.multiplicative_energy_mono Finset.multiplicativeEnergy_mono
-#align finset.additive_energy_mono Finset.additiveEnergy_mono
-
-@[to_additive additiveEnergy_mono_left]
-theorem multiplicativeEnergy_mono_left (hs : s₁ ⊆ s₂) :
-    multiplicativeEnergy s₁ t ≤ multiplicativeEnergy s₂ t :=
-  multiplicativeEnergy_mono hs Subset.rfl
-#align finset.multiplicative_energy_mono_left Finset.multiplicativeEnergy_mono_left
-#align finset.additive_energy_mono_left Finset.additiveEnergy_mono_left
-
-@[to_additive additiveEnergy_mono_right]
-theorem multiplicativeEnergy_mono_right (ht : t₁ ⊆ t₂) :
-    multiplicativeEnergy s t₁ ≤ multiplicativeEnergy s t₂ :=
-  multiplicativeEnergy_mono Subset.rfl ht
-#align finset.multiplicative_energy_mono_right Finset.multiplicativeEnergy_mono_right
-#align finset.additive_energy_mono_right Finset.additiveEnergy_mono_right
-
-@[to_additive le_additiveEnergy]
-theorem le_multiplicativeEnergy : s.card * t.card ≤ multiplicativeEnergy s t := by
+#align finset.multiplicative_energy Finset.mulEnergy
+#align finset.additive_energy Finset.addEnergy
+
+@[inherit_doc Finset.mulEnergy]
+scoped[AddCombi] notation3:max "Eₘ[" s ", " t "]" => Finset.mulEnergy s t
+
+@[inherit_doc Finset.addEnergy]
+scoped[AddCombi] notation3:max "E[" s ", " t "]" => Finset.addEnergy s t
```

Comment:
If these are going to `inherit_doc`, then please mention the notation in the doc for the declarations.

Reply:
I don't really understand this comment: Now that I've mentioned the notation in the docstring, I can't use `inherit_doc` anymore since it would make the notation hover explain how to get the notation working!

Reply:
I think that's a good thing? Just because the scope is open doesn't mean the reader knows which scope is the one responsible. If I'm making a new file, the hover on the notation in your file will tell me how to get access to it myself.